rudder-web/src/main/scala/com/normation/rudder/web/components/RuleGrid.scala

```diff
@@ -542,7 +542,7 @@ class RuleGrid(
     val t0 = System.currentTimeMillis
 
     val nodes = groupsLib.getNodeIds(line.rule.targets, nodesInfo)
-    val (policyMode,explanation) = ComputePolicyMode.ruleMode(globalMode, line.rule.directiveIds.map(directiveLib.allDirectives(_)).map(_._2))
+    val (policyMode,explanation) = ComputePolicyMode.ruleMode(globalMode, line.rule.directiveIds.map(directiveLib.allDirectives.get(_)).flatMap(_.map(_._2)))
```

Comment:
Could you add a comment sating what is the will here? 
Something like:
"We don't fail on missing directive. Most likelly a deleted directive link wasn't updated correctly. We should assure consistency on all rudder config elem, but until that (big) change, it seems to be the most resilient solution".